src/Sylius/Bundle/ApiBundle/Validator/Constraints/ShopUserResetPasswordTokenExistsValidator.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Validator\Constraints;
+
+use Sylius\Component\User\Model\UserInterface;
+use Sylius\Component\User\Repository\UserRepositoryInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+final class ShopUserResetPasswordTokenExistsValidator extends ConstraintValidator
+{
+    /**
+     * @param UserRepositoryInterface<UserInterface> $userRepository
+     */
+    public function __construct(private UserRepositoryInterface $userRepository)
```

Comment:
Just a suggestion, one of those things, that always forces me to double check which user is actually being used.

```suggestion
    public function __construct(private UserRepositoryInterface $shopUserRepository)
```